infra/app/api.bicep

```diff
@@ -14,33 +16,82 @@ param instanceMemoryMB int = 2048
 param identityClientId string = ''
+param enableBlob bool = true
+param enableQueue bool = false
+param enableTable bool = false
+param enableFile bool = false
+
+@allowed(['SystemAssigned', 'UserAssigned'])
+param identityType string = 'UserAssigned'
 
 var applicationInsightsIdentity = 'ClientId=${identityClientId};Authorization=AAD'
+var kind = 'functionapp,linux'
+
+resource stg 'Microsoft.Storage/storageAccounts@2022-09-01' existing = {
+  name: storageAccountName
+}
+
+resource applicationInsights 'Microsoft.Insights/components@2020-02-02' existing = if (!empty(applicationInsightsName)) {
+  name: applicationInsightsName
+}
 
-module api '../core/host/functions-flexconsumption.bicep' = {
-  name: '${serviceName}-functions-module'
+// Create a Flex Consumption Function App to host the API
+module api 'br/public:avm/res/web/site:0.15.1' = {
+  name: '${serviceName}-flex-consumption'
   params: {
+    kind: kind
     name: name
-    location: location
     tags: union(tags, { 'azd-service-name': serviceName })
-    identityType: 'UserAssigned'
-    identityId: identityId
-    appSettings: union(appSettings,
-      {
-        AzureWebJobsStorage__clientId : identityClientId
+    serverFarmResourceId: appServicePlanId
+    managedIdentities: {
+      systemAssigned: identityType == 'SystemAssigned'
+      userAssignedResourceIds: [
+        '${identityId}'
+      ]
+    }
+    functionAppConfig: {
+      location: location
+      deployment: {
+        storage: {
+          type: 'blobContainer'
+          value: '${stg.properties.primaryEndpoints.blob}${deploymentStorageContainerName}'
+          authentication: {
+            type: identityType == 'SystemAssigned' ? 'SystemAssignedIdentity' : 'UserAssignedIdentity'
+            userAssignedIdentityResourceId: identityType == 'UserAssigned' ? identityId : '' 
+          }
+        }
+      }
+      scaleAndConcurrency: {
+        instanceMemoryMB: instanceMemoryMB
+        maximumInstanceCount: maximumInstanceCount
+      }
+      runtime: {
+        name: runtimeName
+        version: runtimeVersion
+      }
+    }
+    siteConfig: {
+      alwaysOn: false
+    }
+    virtualNetworkSubnetId: !empty(virtualNetworkSubnetId) ? virtualNetworkSubnetId : null
+    appSettingsKeyValuePairs: union(appSettings, {
+        // Only include required credential settings unconditionally
+        AzureWebJobsStorage__credential: 'managedidentity'
+        AzureWebJobsStorage__clientId: identityClientId
+        
+        // Include storage endpoint settings conditionally based on feature flags
+        AzureWebJobsStorage__blobEndpoint: enableBlob ? stg.properties.primaryEndpoints.blob : null
+        AzureWebJobsStorage__queueEndpoint: enableQueue ? stg.properties.primaryEndpoints.queue : null
+        AzureWebJobsStorage__tableEndpoint: enableTable ? stg.properties.primaryEndpoints.table : null
+        AzureWebJobsStorage__fileEndpoint: enableFile ? stg.properties.primaryEndpoints.file : null
+        
```

Comment:
```suggestion
        AzureWebJobsStorage__blobServiceUri: enableBlob ? stg.properties.primaryEndpoints.blob : null
        AzureWebJobsStorage__queueServiceUri: enableQueue ? stg.properties.primaryEndpoints.queue : null
        AzureWebJobsStorage__tableServiceUri: enableTable ? stg.properties.primaryEndpoints.table : null
        AzureWebJobsStorage__fileServiceUri: enableFile ? stg.properties.primaryEndpoints.file : null
        
```

Reply:
Thanks heard your verbal feedback and already fixed.